File: ts/example/src/tests/query_doubles.ts
import {ExampleQuery} from "../query";


export class ExampleQueryStub implements ExampleQuery {
  constructor(public nextResult: string) {}


  fetchSomething(): string {
    return this.nextResult;
  }
}


// NOTE: You can write ExampleQuerySpy, but it is not recommended.
//       Because Queries should not have any indirect output.
//       So spies for query should not be needed.

File: swift/example/Tests/VanillaMockExampleTests/QueryDoubles.swift
@testable import VanillaMockExample



class ExampleQueryStub: ExampleQueryProtocol {
    public var nextResult: String


    init(willReturn nextResult: String) {
        self.nextResult = nextResult
    }


    func fetchSomething() -> String {
        self.nextResult
    }
}


// NOTE: You can write ExampleQuerySpy, but it is not recommended.
//       Because Queries should not have any indirect output.
//       So spies for query should not be needed.